Add in-cluster config support

mbohlool opened the pull request:
enables client-python to be used inside a kubernetes cluster.

Fixes #26

marun commented:
What's the plan for enabling automated testing (unit/integration/e2e)?  There's only so much a review of code can do, and I'd rather be reviewing code + tests.

mbohlool commented:
@marun Thanks for demanding test from a lazy engineer here :) It is in much better shape now.
Refactored code to be able to test and added tests. PTAL.

marun commented on kubernetes/config/incluster_config_test.py:
```diff
@@ -0,0 +1,101 @@
+
+from .incluster_config import ConfigException, InClusterConfigLoader
+
+_TEST_PORT_ENV_NAME = "port"
```
Consider reusing the names of the env vars defined as constants in the module under test (e.g. ``_SERVICE_HOST_ENV_NAME``) rather than defining new names in this test module.  Since the environ dict is passed to the function under test, there is no need to vary the names of the env vars and the names can be fixed across runtime and test.

mbohlool commented:
@marun I am going to discuss ci and e2e with kubernetes e2e test folks in a separate thread/issue.

mbohlool commented on kubernetes/config/incluster_config.py:
```diff
@@ -0,0 +1,76 @@
+                                         self._environ[self._port_env_name]))
+
+        if not os.path.isfile(self._token_filename):
+            raise ConfigException("Service token file does not exists.")
```
I've used the InClusterConfigLoader to store the values and separate load and set methods.

mbohlool commented on kubernetes/config/incluster_config.py:
```diff
@@ -0,0 +1,76 @@
+        if not os.path.isfile(self._token_filename):
+            raise ConfigException("Service token file does not exists.")
+
+        with open(self._token_filename, 'r') as f:
```
Done.

mbohlool commented:
@marun, I've addressed all of your comments. As all of them were optional, I assume this PR is look good. Please apply LGTM next time if your comments are all optional. Thanks for the review.

marun commented on kubernetes/config/incluster_config.py:
```diff
@@ -0,0 +1,86 @@
+            raise ConfigException("Service token file does not exists.")
+
+        with open(self._token_filename) as f:
+            self.token = f.read()
```
What if the file is empty?

mbohlool replied:
Missed these comments, addressed them in #50

marun commented on kubernetes/config/incluster_config.py:
```diff
@@ -0,0 +1,86 @@
+        self._set_config()
+
+    def _load_config(self):
+        if (self._host_env_name not in self._environ or
```
What about the case where either of the env vars are defined but empty (e.g. ``KUBERNETES_SERVICE_HOST=``)?

mbohlool replied:
Missed these comments, addressed them in #50
